File: paypal/lib/core/models/marketplace_job.dart
class MarketplaceJob {
  const MarketplaceJob({
    required this.id,
    required this.title,
    required this.description,
    required this.budgetUsd,
    this.clientUserId,
    this.status,
    this.createdAt,
    this.checkoutOrderId,
  });

  final String id;
  final String title;
  final String description;
  final double budgetUsd;
  final String? clientUserId;
  final String? status;
  final DateTime? createdAt;
  final String? checkoutOrderId;

  factory MarketplaceJob.fromJson(Map<String, dynamic> json) {
    return MarketplaceJob(
      id: json['id'] as String? ?? '',
      title: json['title'] as String? ?? '',
      description: json['description'] as String? ?? '',
      budgetUsd: (json['budgetUsd'] as num?)?.toDouble() ?? 0,
      clientUserId: json['clientUserId'] as String?,
      status: json['status'] as String?,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
      checkoutOrderId: json['checkoutOrderId'] as String?,
    );
  }

  MarketplaceJob copyWith({String? checkoutOrderId, String? status}) {
    return MarketplaceJob(
      id: id,
      title: title,
      description: description,
      budgetUsd: budgetUsd,
      clientUserId: clientUserId,
      status: status ?? this.status,
      createdAt: createdAt,
      checkoutOrderId: checkoutOrderId ?? this.checkoutOrderId,
    );
  }
}

File: paypal-backend/src/main/java/com/paypal/backend/entity/MarketplaceJob.java
package com.paypal.backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "marketplace_jobs")
@Getter
@Setter
public class MarketplaceJob {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    UUID clientUserId;

    String title;

    String description;

    BigDecimal budgetUsd;

    UUID checkoutOrderId;

    @Enumerated(EnumType.STRING)
    JobStatus status;

    LocalDateTime createdAt;
}
